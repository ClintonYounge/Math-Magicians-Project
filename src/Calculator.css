* {
    box-sizing: border-box;
    list-style-type: none;
}

body {
    background-color: black;
  margin: 0;
  font-family:
    -apple-system,
    BlinkMacSystemFont,
    'Segoe UI',
    'Roboto',
    'Oxygen',
    'Ubuntu',
    'Cantarell',
    'Fira Sans',
    'Droid Sans',
    'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family:
    source-code-pro,
    Menlo,
    Monaco,
    Consolas,
    'Courier New',
    monospace;
}

main {
    width: 80%;
    margin: auto;
    margin-top: 5%;
}
.show-calculations {
    margin: auto;
    border: 1px solid white;
    background-color: grey;
    height: 15vh;
    text-align: end;
}

.show-calculations p {
    width: 98%;
}

.input-rows {
    height: 100%;
    background-color: whitesmoke;
}

.inputRow-1,
.inputRow-2,
.inputRow-3,
.inputRow-4,
.inputRow-5 {
    display: flex;
    margin: auto;
    height: 13VH;
    justify-content: space-between;
    list-style-position: inside;
    padding-left: 0;
}

li {
    width: 100%;
    text-align: center;
    border: 1px solid grey;
    padding-top: 3%;
}

.orangeBG {
    background-color: orange;
}

.R5-li2,
.R5-li3 {
    width: 50%;
}
