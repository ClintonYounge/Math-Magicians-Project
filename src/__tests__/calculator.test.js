import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Calculator from '../components/calculator';

describe('Calculator interaction', () => {
  test('updates result when button 3 is clicked', () => {
    // Arrange
    const { getByText, getByTestId } = render(<Calculator />);

    // Act
    fireEvent.click(getByText('3'));

    // Assert
    const calculatorScreen = getByTestId('outputResult');
    expect(calculatorScreen).toHaveTextContent('3');
  });

  test('updates outputResult when buttons 2 + 4 and = are clicked', () => {
    // Arrange
    const { getByText, getByTestId } = render(<Calculator />);
    // Act
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('+'));
    fireEvent.click(getByText('4'));
    fireEvent.click(getByText('='));
    // Assert
    const calculatorScreen = getByTestId('outputResult');
    expect(calculatorScreen).toHaveTextContent('6');
  });

  test('updates outputResult when buttons 2 x 4 and = are clicked', () => {
    // Arrange
    const { getByText, getByTestId } = render(<Calculator />);
    // Act
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('x'));
    fireEvent.click(getByText('4'));
    fireEvent.click(getByText('='));
    // Assert
    const calculatorScreen = getByTestId('outputResult');
    expect(calculatorScreen).toHaveTextContent('8');
    expect(calculatorScreen).toMatchSnapshot();
  });
});
