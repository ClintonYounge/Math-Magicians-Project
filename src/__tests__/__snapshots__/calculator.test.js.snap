// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator interaction updates outputResult when buttons 2 x 4 and = are clicked 1`] = `
<div
  class="show-calculations"
  data-testid="outputResult"
  id="show-calculations"
>
  <div
    class="component-display"
  >
    <div>
      8
    </div>
  </div>
</div>
`;
